src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.ipp

```diff
@@ -0,0 +1,1116 @@
+        default:
+        {
+            ret = RETCODE_BAD_PARAMETER;
+            break;
+        }
+    }
+
+    return ret;
+}
+
+ReturnCode_t array_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& array_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_ARRAY)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not an array.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    DynamicType::_ref_type internal_type;
+    ret = container_internal_type(dyn_type, internal_type);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    ret = type_kind_to_str(internal_type, array_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    for (const auto& bound : bounds)
+    {
+        array_str += "[";
+        array_str += std::to_string(bound);
+        array_str += "]";
+    }
+
+    return ret;
+}
+
+ReturnCode_t map_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& map_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_MAP)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a map.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::string key_str;
+    const auto key_type = type_descriptor->key_element_type();
+    ret = type_kind_to_str(key_type, key_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::string value_str;
+    const auto value_type = type_descriptor->element_type();
+    ret = type_kind_to_str(value_type, value_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::stringstream ss;
+
+    ss << "map<" << key_str << ", " << value_str << ">";
+
+    map_str = ss.str();
+
+    return ret;
+}
+
+ReturnCode_t sequence_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& sequence_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_SEQUENCE)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a sequence.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    DynamicType::_ref_type internal_type;
+    ret = container_internal_type(dyn_type, internal_type);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    ret = type_kind_to_str(internal_type, sequence_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    sequence_str = "sequence<" + sequence_str;
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    for (const auto& bound : bounds)
```

Comment:
I don't think it's possible to have multiple dimensions in sequences.

Reply:
Also, I find this logic a bit odd, in the sense that I believe the only case where UNBOUNDED may be present it's in the first (unique) dimension.